[Image Embeddable] Integrate with drilldowns

Dosant opened the pull request:
## Summary

Close https://github.com/elastic/kibana/issues/148008
follow up to https://github.com/elastic/kibana/pull/146421

Adds drilldowns integration to image embeddable 
Now it is possible to create Dashboard-to-Dashboard or URL to an image. The drilldown is triggered on image click. 



https://user-images.githubusercontent.com/7784120/209163963-01ddf17d-fa68-465a-876b-05b92260cae1.mov

Dosant commented on src/plugins/image_embeddable/public/image_embeddable/image_embeddable.tsx:
```diff
@@ -48,6 +54,10 @@ export class ImageEmbeddable extends Embeddable<ImageEmbeddableInput> {
     const input = useObservable(props.embeddable.getInput$(), props.embeddable.getInput());
 
+    React.useLayoutEffect(() => {
+      import('./image_embeddable_lazy');
```
Importing `scss` directly added 5kb to the main bundle because of all the scss runtime helpers

vadimkibana reviewed (approved):
Code LGTM.

vadimkibana commented:
I've ran the PR locally, looks really good. Some thoughts:

- Maybe when link is added the mouse cursor should be `pointer`.
- The license on URL drilldown is higher than other drilldowns, I wonder if that is OK for the Image Embeddable.

Dosant commented:
@vadimkibana, 

> The license on URL drilldown is higher than other drilldowns, I wonder if that is OK for the Image Embeddable.

I think this is OK. and consistent :) 

> Maybe when link is added the mouse cursor should be pointer.

I also think so, Initially I implemented this, but reverted because of some quirks. Here I was removing it: https://github.com/elastic/kibana/pull/147995/commits/40341fbd21e0ad1b358ce36f54e67562b0af9b90

Here is why I reverted: 

- I had to execute `getTriggerCompatibleActions` on each embeddable panel re-render. I though that maybe the overhead could be not worse it for the panel 
- When saving a drilldown re-render didn't happen. It happens only on switching from edit to view mode. So it seemed it could be confusing that image wasn't clickable straightaway when drilldown was added. I decided instead of fixing it - remove this. 

I am thinking I should get it back and not worry about excessive `getTriggerCompatibleActions` calls, wdyt?

Dosant commented:
@elasticmachine merge upstream

ThomThomson reviewed (approved):
Presentation team changes LGTM! Code review + ran locally and made a few image drilldowns.  The pointer change is a nice touch!

Dosant commented:
@elasticmachine merge upstream

mdefazio reviewed (commented):
The only piece that feels a bit off for me is the copy on the trigger selection:

<img width="619" alt="image" src="https://user-images.githubusercontent.com/3756330/210828097-fe24082b-81ba-4bb2-8260-3891adba3ea8.png">

- - - 


The descriptions below each option feel redundant to the label.  Perhaps the following as another option:

**Context menu**
A new action will be added to the panel's context menu

**Image**
Clicking the image will trigger the action

Perhaps there were previous discussions around the copy that I missed.  Maybe not a blocker, but wanted to offer another option.

Dosant commented:
> The only piece that feels a bit off for me is the copy on the trigger selection:
> 
> The descriptions below each option feel redundant to the label. Perhaps the following as another option:
> 
> **Context menu** A new action will be added to the panel's context menu
> 
> **Image** Clicking the image will trigger the action
> 
> Perhaps there were previous discussions around the copy that I missed. Maybe not a blocker, but wanted to offer another option.

Thanks! I like this suggestion and applied the change

mdefazio reviewed (approved):
Thanks for updating!
